src/zapv2/replacer.py

```diff
@@ -35,12 +35,14 @@ def rules(self):
         """
         return six.next(six.itervalues(self.zap._request(self.zap.base + 'replacer/view/rules/')))
 
-    def add_rule(self, description, enabled, matchtype, matchregex, matchstring, replacement, initiators=None, apikey=''):
+    def add_rule(self, description, enabled, matchtype, matchregex, matchstring, replacement=None, initiators=None, apikey=''):
         """
         Adds a replacer rule. For the parameters: desc is a user friendly description, enabled is true or false, matchType is one of [REQ_HEADER, REQ_HEADER_STR, REQ_BODY_STR, RESP_HEADER, RESP_HEADER_STR, RESP_BODY_STR], matchRegex should be true if the matchString should be treated as a regex otherwise false, matchString is the string that will be matched against, replacement is the replacement string, initiators may be blank (for all initiators) or a comma separated list of integers as defined in <a href="https://github.com/zaproxy/zaproxy/blob/develop/src/org/parosproxy/paros/network/HttpSender.java">HttpSender</a>  
         This component is optional and therefore the API will only work if it is installed
         """
-        params = {'description': description, 'enabled': enabled, 'matchType': matchtype, 'matchRegex': matchregex, 'matchString': matchstring, 'replacement': replacement, 'apikey': apikey}
+        params = {'description': description, 'enabled': enabled, 'matchType': matchtype, 'matchRegex': matchregex, 'matchString': matchstring, 'apikey': apikey}
+        if replacement is not None:
```

Comment:
Does this work?

```suggestion
        if not replacement:
```

This condition would be met if `replacement` evalutes to `False` or `None`. I assume that would work but I don't have enough context about how it gets used.

Reply:
That would change the desired behaviour, we want to send the parameter whenever it has a value (even if false).